sklearn/svm/src/libsvm/svm.cpp

```diff
@@ -3101,6 +3101,44 @@ const char *PREFIX(check_parameter)(const PREFIX(problem) *prob, const svm_param
 		free(count);
 	}
 
+	if(svm_type == C_SVC ||
+	   svm_type == EPSILON_SVR ||
+	   svm_type == NU_SVR ||
+	   svm_type == ONE_CLASS)
+	{
+	    PREFIX(problem) newprob;
+          // filter samples with negative weights like algorithm PREFIX(train)
+	    remove_zero_weight(&newprob, prob);
+
+	    char* msg = NULL;
+          // no samples were removed from training set - all weights positive.
+	    if(prob->l == newprob.l)
+	        msg = NULL;
+          // all samples were removed
+	    else if(newprob.l == 0)
+	        msg =  "Invalid input - all samples have zero/negative weights.";
+        else if(svm_type == C_SVC)
```

Comment:
I was very confused by the indentation here.

Reply:
Sorry about that. Missed it. It is ifxed now.